### ISSUE TITLE
Make initial client side blazor app download faster

### ISSUE THREAD

**Issue by jucchytil** [aspnet/AspNetCore#10545]
The initial download time for users is too long.

During the build process, analyze the entire app and output the minimal number of files needed to get the app to display in a browser.  

Perhaps (just ideas here)
- All executables (wasm/dlls) would only link in the absolutely needed functionality and be compiled to a single wasm file
- minify multiple css and js files to single files if possible
- Explore combining to a single file similar to .net core 3.0 single file executables
- Explore combining to a zip file, download zip file to browser and use run-from-zip type capability similar to what can be used in azure-webapps/functions.

In short. make the user experience feel as fast as a well designed javascript based app.

**Comment by jucchytil** [comment]
If anyone knows of a document that describes best practices to improve the user experience during the initial client side blazor download, I would be grateful for links to read.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @jucchytil.
This is a never-ending journey for us and we're constantly working on improving the startup performance of a client-side blazor application. The main contributor to this at the moment is the size of the `mono.wasm`, which we're working on with the `https://github.com/mono/mono` folks to improve.
We also have another issue, where we plan to enable lazy-loading of Blazor applications in the future: https://github.com/aspnet/AspNetCore/issues/5465

**Comment by jucchytil** [comment]
would it make  sense to compile the mon.wasm together with all the other executables into one wasm file during the build and then just download the one wasm file?
